src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -302,11 +301,13 @@ internal void StartMessageBatchReceiver(CancellationToken cancellationToken)
                                     }
                                 }
                             }
-
-                            // Close the session and release the session lock
-                            if (_isSessionsEnabled)
+                            else
                             {
-                                await receiver.CloseAsync();
+                                // Close the session and release the session lock

```

Comment:
Would be good to enhance this comment and explain that we are only closing the session after draining all prefetched messages.

Reply:
Comment is enhanced